README.md

```diff
@@ -19,14 +19,188 @@ The cleaned must-gather can then be found in the `must-gather-output-cleaned` fo
+
+Another configuration flag that we support for each obfuscation type is the `target`. The target is useful when the confidential information can be found not only in the file content, but also in the folder or file names. 
+This can very frequently happen with IP addresses, for example through node names. You can control that independently for each type like this:
+
+```
+config:
+  obfuscate:
+  - type: MAC
+    replacementType: Consistent
+    target: FileContents
+  - type: IP
+    replacementType: Consistent
+    target: FilePath
+```
+
+As you can see, the MAC obfuscator would work on file content whereas the IP obfuscator would work only on FilePaths. There is a mixed target called `All`, that will obfuscate on both paths and contents. 
+The default if no target is specified is `FileContents`, it is thus always recommended using the IP obfuscator with `target: All` to not accidentally leak IP information through folder names.
+
+The third built-in type of obfuscation is `Domain`, let's take a look how this can be configured:
+
+```
+config:
+  obfuscate:
+  - type: Domain
+    domainNames:
+    - "rhcloud.com"
+    - "dev.rhcloud.com"
+```
+
+As you can see, this type must be customized by supplying domain names. 
+Kubernetes resources are defined along with their domain names (eg "apps.openshift.io/v1") and thus would be automatically recognized as such and obfuscated as a false-positive.
+We thus kindly ask the user to supply their confidential domain names manually through the configuration.
+
+The above definition will obfuscate `rhcloud.com` as `domain0000001` (consistent) or as `xxxxxxxxxxxxx` (static). 
+Note that this does not include subdomains, they would need to be separately obfuscated. 
+A domain name defined as `staging.rhcloud.com` would only be obfuscated as `staging.domain0000001`, thus, it makes sense to also include all common subdomains as done in the example with "dev".
+
+### Custom Obfuscations
+
+Aside from the above three built-in types to obfuscate, we also offer custom obfuscators that allow users to fine-tune the replacement of certain strings (eg custom auth token formats, confidential domain knowledge or keywords).
+This comes through two custom types, one is replacement via type `Keywords` the other via type `Regex`.
+
+#### Keywords
+
+Let's start by looking into `Keywords` first:
+
+```
+config:
+  obfuscate:
+  - type: Keywords
+    replacement:
+       hello: bye
+       tomorrow: yesterday
+```
+
+This configuration will simply replace the strings on the left-hand side with the values on the right-hand side. 
+The `target` variable here is supported as well, so you can also target specific files and obfuscate their name, for example:
+
+```
+config:
+  obfuscate:
+  - type: Keywords
+    target: FilePath
```

Comment:
```suggestion
    target: <FilePath>
```
This would help users to understand that it's a placeholder and distinguish from static words like `Keywords`